pkg/envoy/cds/cluster.go

```diff
@@ -128,7 +94,72 @@ func getUpstreamServiceCluster(downstreamIdentity identity.ServiceIdentity, upst
 			},
 		}
 	}
+	return remoteCluster, nil
+}
+
+// getGatewayUpstreamServiceCluster returns an Envoy Cluster corresponding to the given upstream service for the multicluster gateway
+func getGatewayUpstreamServiceCluster(upstreamSvc service.MeshService, cfg configurator.Configurator, ports []corev1.ServicePort) (*xds_cluster.Cluster, error) {
```

Comment:
Can we not reuse `getUpstreamServiceCluster` and use `ClusterOption` to override for multicluster? It would avoid code duplication.

Reply:
I looked into this, reusing `getUpstreamServiceCluster` will not be very effective and will add another condition. The name of the remote clusters for the gateway will be the entire server name of the service which makes it different from `getUpstreamServiceCluster`. Further the cluster discovery mechanism here will be strict DNS and we add the endpoints making it very different from the condition that exists in `getUpstreamServiceCluster`. Adding another method in my opinion is cleaner for this scenario.